pkg/packerjanitor/packer_janitor.go

```diff
@@ -0,0 +1,242 @@
+package packerjanitor
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"go.uber.org/zap"
+
+	"time"
+)
+
+const (
+	// RFC8601 is the date/time format used by AWS.
+	RFC8601 = "2006-01-02T15:04:05.000Z"
+	// DryRun is the type of error thrown by AWS when a task fails
+	// because it was run with the DryRun option but would have
+	// otherwise succeeded.
+	DryRun = "DryRunOperation"
+)
+
+// PackerClean is a generic struct used for the various functions.
+type PackerClean struct {
+	Delete         bool
+	ExpirationDate time.Time
+	Logger         *zap.Logger
+	EC2Client      ec2iface.EC2API
+}
+
+// GetPackerInstances -- find all running instances that are Packer
+// builds older than X and returns them in a list
+func (p *PackerClean) GetPackerInstances() ([]*ec2.Instance, error) {
+	var output *ec2.DescribeInstancesOutput
+	// Instances Packer starts all have the Name tag "Packer Builder".
+	packerFilter := &ec2.Filter{
+		Name:   aws.String("tag:Name"),
+		Values: []*string{aws.String("Packer Builder")},
+	}
+	input := &ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{packerFilter},
+	}
+	output, err := p.EC2Client.DescribeInstances(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				p.Logger.Error("Encountered AWS error attempting to get instance list",
+					zap.Error(aerr),
+				)
+			}
+		} else {
+			p.Logger.Error("Error while attempting to get instance list",
+				zap.Error(err),
+			)
+			return nil, err
+		}
+	}
+
+	// The output gives us reservations; we need to get the actual
+	// instances out of them, and look to make sure they are older
+	// than the time we're looking for.
+	var instanceList = []*ec2.Instance{}
```

Comment:
This is totally valid Go, but generally ``` var instanceList []*ec2.Instance```, without the ```=``` and ```{}```, is what gophers prefer. I've seen you write it the other way, so it's probably just a typo here!

Reply:
I suspect it's just me not remembering what I did last time...I'm still at the point where I'm relearning lessons over again when I do something in go, so :)